Convert type comments to annotations in torch/nn

stroxler opened the pull request:
Summary:
This commit was produced by running
```
python -m libcst.tool codemod --no-format --jobs=1 convert_type_comments.ConvertTypeComments caffe2/torch/nn/ --no-quote-annotations
```
and then manually fixing unreadable lines by breaking up very
long function defintiion (unfortuantely
it's very difficult to fully automate tranforms of code that
isn't autoformatted).

Test Plan:
Wait for CI. This should be safe, the types all appear to be valid - but it's
always good to let the jit tests run, in some cases we find typing errors that
crash tests.

release notes:

Converts type comments to annotations.

topics:

Refactor

Differential Revision: D34147388

albanD reviewed (approved):
Thanks for the update!

jbschlosser reviewed (approved):
Cool :)

stroxler committed:
Convert type comments to annotations in torch/nn (#72662)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/72662

This commit was produced by running
```
python -m libcst.tool codemod --no-format --jobs=1 convert_type_comments.ConvertTypeComments caffe2/torch/nn/ --no-quote-annotations
```
and then manually fixing unreadable lines by breaking up very
long function defintiion (unfortuantely
it's very difficult to fully automate tranforms of code that
isn't autoformatted).

Test Plan:
Wait for CI. This should be safe, the types all appear to be valid - but it's
always good to let the jit tests run, in some cases we find typing errors that
crash tests.

Reviewed By: jbschlosser, albanD

Differential Revision: D34147388

fbshipit-source-id: c2fdeae43ef82209d053555c592a8e31f66e5d08

stroxler commented:
Thanks for the reviews!

Rebase, fix a merge conflict (a type was changed this morning)
